subsys/bluetooth/host/hci_core.c

```diff
@@ -516,6 +516,8 @@ static int set_le_adv_enable_legacy(bool enable)
 		return err;
 	}
 
+	atomic_set_bit_to(bt_dev.flags, BT_DEV_ADVERTISING, enable);
```

Comment:
Remove this.
Should be covered by this:
https://github.com/zephyrproject-rtos/zephyr/pull/24315/files#diff-8c3bce98d0cfacce4012133a2290e094R511-R512

Reply:
Thanks joerchan,

I understand that the BT_DEV_ADVERTISING flag represents the controller state (the comment above the enum definition in hci_core.h), so I didn't get the point: what is the expected race condition ?

Reply:
Updated the link.

You can see there reason for this here: https://github.com/zephyrproject-rtos/zephyr/commit/0bf9931c2c708ab289dd4800afb82d7865e04dc9